lld/ELF/Thunks.cpp

```diff
@@ -578,6 +622,48 @@ void AArch64ADRPThunk::addSymbols(ThunkSection &isec) {
   addSymbol("$x", STT_NOTYPE, 0, isec);
 }
 
+void AArch64BTILandingPadThunk::addSymbols(ThunkSection &isec) {
+  addSymbol(saver().save("__AArch64BTIThunk_" + destination.getName()),
+            STT_FUNC, 0, isec);
+  addSymbol("$x", STT_NOTYPE, 0, isec);
+}
+
+InputSection *AArch64BTILandingPadThunk::getTargetInputSection() const {
```

Comment:
This function is not needed.

Relocations.cpp calls `ts = getISThunkSec(targetsec)` without checking `getTargetInputSection`.

Reply:
It is called in `mergeCmp()` but I've removed it anyway as it is not a strict requirement that the landing pads are before the target section, they just need to be within direct branch range.